### PR TITLE
(feat teaming): API schema and Database schema changes

### PR DESCRIPTION
**Pull request by Sanjay1611**
Signed-off-by: Sanjay Nathani <sanjay.nathani@mayadata.io>

## Proposed changes

This PR contain following changes

- slight modification in the structure of code
- teaming implementation on the database
- project database schema modification. It will look like this
```
{
  "_id": "3f596927-9704-4fa3-80aa-449509ed953c",
  "username": "sanjay1234",
  "email": "sanjay@gmail.com",
  "is_email_verified": null,
  "company_name": null,
  "name": "Sanjay Nathani",
  "role": null,
  "state": null,
  "created_at": "2020-08-27 15:41:11.0053204 +0530 IST m=+423.016224229",
  "updated_at": "",
  "removed_at": ""
}
```
- modification in `getUser` and `getProject` queries

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/backend/graphql-server/go.mod [hunk 1]
```diff
@@ -4,7 +4,13 @@ go 1.14
 
 require (
 	github.com/99designs/gqlgen v0.11.3
+	github.com/Azure/azure-sdk-for-go v45.1.0+incompatible // indirect
+	github.com/Azure/go-autorest/autorest v0.11.4 // indirect
+	github.com/Azure/go-autorest/autorest/adal v0.9.2 // indirect
+	github.com/Azure/go-autorest/autorest/to v0.4.0 // indirect
+	github.com/Azure/go-autorest/autorest/validation v0.3.0 // indirect
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
+	github.com/globalsign/mgo v0.0.0-20181015135952-eeefdecb41b8 // indirect
```
Comment (gdsoumya) [inline litmus-portal/backend/graphql-server/go.mod:13]: Why are we again using mgo? also what are the Azure packages required for?

File: litmus-portal/backend/graphql-server/pkg/database/mongodb/operations/project.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database/mongodb"
+	dbSchema "github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database/mongodb/schema"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var projectCollection *mongo.Collection
+
+func init() {
```
Comment (gdsoumya) [inline litmus-portal/backend/graphql-server/pkg/database/mongodb/operations/project.go:15]: Init all the collections in one place in the `initAllCollections` functions, also declare all the collections in the init file.

File: litmus-portal/backend/graphql-server/pkg/database/mongodb/operations/usermanagement.go [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
-	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/graph/model"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database/mongodb"
+	dbSchema "github.com/litmuschaos/litmus/litmus-portal/backend/graphql-server/pkg/database/mongodb/schema"
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/mongo"
 )
 
 var userCollection *mongo.Collection
 
+func init() {
```
Comment (gdsoumya) [inline litmus-portal/backend/graphql-server/pkg/database/mongodb/operations/usermanagement.go:15]: Same comment as project

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: litmus-portal/backend/graphql-server/go.mod (hunk 1 above)
Comment (gdsoumya) [reply litmus-portal/backend/graphql-server/go.mod:13]: Can you do a `go mod tidy` to remove unused packages

File: litmus-portal/backend/graphql-server/pkg/database/mongodb/operations/project.go (hunk 2 above)
Comment (Sanjay1611) [reply litmus-portal/backend/graphql-server/pkg/database/mongodb/operations/project.go:15]: In this approach it wouldn't be required to have all the collections in the init file. We will just have to create a separate file in order to add any resource

**Review by rajdas98 — APPROVED** [review]

**Pull request merged**